From cb346befc55f4831fca2a2278267344ee3f9ff6a Mon Sep 17 00:00:00 2001
From: Venkateshwar Rao G <venkateshwar.rao.gannavarapu@xilinx.com>
Date: Thu, 8 Feb 2018 15:09:02 -0800
Subject: [PATCH 0260/1851] drm: xlnx: bridge: Add support for set and get
 output parameters

commit 27e634023ba30723e17bed390e41aff5aeaab45f from
https://github.com/Xilinx/linux-xlnx.git

This adds the support for set and get output parameters like width,
height and color format for VPSS color space converter and scaler.

Signed-off-by: Venkateshwar Rao G <venkateshwar.rao.gannavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_bridge.h | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_bridge.h b/drivers/gpu/drm/xlnx/xlnx_bridge.h
index 8af32b53d23b..2dd5ca196a9a 100644
--- a/drivers/gpu/drm/xlnx/xlnx_bridge.h
+++ b/drivers/gpu/drm/xlnx/xlnx_bridge.h
@@ -30,6 +30,8 @@ struct xlnx_bridge_debugfs_file;
  * @disable: callback to disable the bridge
  * @set_input: callback to set the input
  * @get_input_fmts: callback to get supported input formats.
+ * @set_output: callback to set the output
+ * @get_output_fmts: callback to get supported output formats.
  * @debugfs_file: for debugfs support
  */
 struct xlnx_bridge {
@@ -42,6 +44,10 @@ struct xlnx_bridge {
 			 u32 width, u32 height, u32 bus_fmt);
 	int (*get_input_fmts)(struct xlnx_bridge *bridge,
 			      const u32 **fmts, u32 *count);
+	int (*set_output)(struct xlnx_bridge *bridge,
+			  u32 width, u32 height, u32 bus_fmt);
+	int (*get_output_fmts)(struct xlnx_bridge *bridge,
+			       const u32 **fmts, u32 *count);
 	struct xlnx_bridge_debugfs_file *debugfs_file;
 };
 
@@ -65,6 +71,10 @@ int xlnx_bridge_set_input(struct xlnx_bridge *bridge,
 			  u32 width, u32 height, u32 bus_fmt);
 int xlnx_bridge_get_input_fmts(struct xlnx_bridge *bridge,
 			       const u32 **fmts, u32 *count);
+int xlnx_bridge_set_output(struct xlnx_bridge *bridge,
+			   u32 width, u32 height, u32 bus_fmt);
+int xlnx_bridge_get_output_fmts(struct xlnx_bridge *bridge,
+				const u32 **fmts, u32 *count);
 struct xlnx_bridge *of_xlnx_bridge_get(struct device_node *bridge_np);
 void of_xlnx_bridge_put(struct xlnx_bridge *bridge);
 
@@ -115,6 +125,22 @@ static inline int xlnx_bridge_get_input_fmts(struct xlnx_bridge *bridge,
 	return 0;
 }
 
+static inline int xlnx_bridge_set_output(struct xlnx_bridge *bridge,
+					 u32 width, u32 height, u32 bus_fmt)
+{
+	if (bridge)
+		return -ENODEV;
+	return 0;
+}
+
+static inline int xlnx_bridge_get_output_fmts(struct xlnx_bridge *bridge,
+					      const u32 **fmts, u32 *count)
+{
+	if (bridge)
+		return -ENODEV;
+	return 0;
+}
+
 static inline struct xlnx_bridge *
 of_xlnx_bridge_get(struct device_node *bridge_np)
 {
-- 
2.31.1

